internal/devices/drivers.go

```diff
@@ -0,0 +1,114 @@
+// +build windows
+
+package devices
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/Microsoft/hcsshim/internal/oci"
+	"github.com/Microsoft/hcsshim/internal/resources"
+	"github.com/Microsoft/hcsshim/internal/uvm"
+	"github.com/pkg/errors"
+)
+
+// getAssignedDeviceUtilityTool gets the path to device-util on the server host
+func getAssignedDeviceUtilityTool(annotations map[string]string) (string, error) {
+	tools, ok := annotations[oci.AnnotationAssignedDeviceUtilityTool]
+	if !ok || tools == "" {
+		return "", errors.New("no driver tools were specified")
+	}
+	if _, err := os.Stat(tools); err != nil {
+		return "", errors.Wrapf(err, "failed to find device installation tools at %s", tools)
+	}
+	return tools, nil
+}
+
+// getAssignedDeviceKernelDrivers gets any device drivers specified on the spec.
+// Drivers are optional, therefore do not return an error if none are on the spec.
+//
+// Paths specified by this annotation should point to directories containing all
+// files and information needed to install given driver(s). This may include .sys,
+// .inf, .cer, and/or other files used during standard installation with pnputil.
+func getAssignedDeviceKernelDrivers(annotations map[string]string) ([]string, error) {
+	csDrivers, ok := annotations[oci.AnnotationAssignedDeviceKernelDrivers]
+	if !ok || csDrivers == "" {
+		return nil, nil
+	}
+	drivers := strings.Split(csDrivers, ",")
+	for _, driver := range drivers {
+		if _, err := os.Stat(driver); err != nil {
+			return nil, errors.Wrapf(err, "failed to find path to drivers at %s", driver)
+		}
+	}
+	return drivers, nil
+}
+
+// setupDeviceUtilTool finds the utility tool's host path, mounts it using vsmb, and
+// returns the UVM path to the tools
+func setupDeviceUtilTool(ctx context.Context, vm *uvm.UtilityVM, annotations map[string]string, closers []resources.ResourceCloser) (string, error) {
```

Comment:
I think we can probably get rid of this function, or at least make it trivial. If we remove the device tool annotation, then this function should just receive the path to the device tool directly, rather than the annotation map. So all this function is doing is calling `addVSMBToUVM`, which can be inlined instead of making this a separate function.